regulations/static/regulations/js/source/views/comment-view.js

```diff
@@ -35,25 +35,84 @@ var CommentView = Backbone.View.extend({
         scrollTop: this.$comment.offset().top
       }, 2000);
     }
+  },
 
+  getStorage: function() {
+    return JSON.parse(window.localStorage.getItem(this.key) || '{}');
   },
 
-  submit: function(e) {
-    e.preventDefault();
-    window.localStorage.setItem(
-      this.key,
-      JSON.stringify({
-        comment: this.$comment.val()
-      })
+  setStorage: function() {
+    var payload = {
+      comment: this.$comment.val(),
+      files: this.$queued.find('.queue-item').map(function(idx, elm) {
+        var $elm = $(elm);
+        return {
+          key: $elm.data('key'),
+          name: $elm.text()
+        };
+      }).get()
+    };
+    window.localStorage.setItem(this.key, JSON.stringify(payload));
+  },
+
+  addQueueItem: function(key, name) {
+    this.$queued.append(
+      $('<div class="queue-item" data-key="' + key + '">' + name + '</div>')
     );
   },
 
   load: function() {
-    var payload = JSON.parse(window.localStorage.getItem(this.key) || '{}');
-    if (payload.comment) {
-      this.$comment.val(payload.comment);
-    }
+    var payload = this.getStorage();
+    this.$comment.val(payload.comment);
+    _.each(payload.files || [], function(file) {
+      this.addQueueItem(file.key, file.name);
+    }.bind(this));
+  },
+
+  addAttachment: function(e) {
+    var file = e.target.files[0];
+    if (!file) { return; }
+    $.when(getUploadUrl(), readFile(file)).done(function(url, data) {
+      $.ajax({
+        type: 'PUT',
+        url: url.url,
+        data: data,
+        contentType: 'application/octet-stream',
+        processData: false
+      }).done(function(resp) {
+        this.addQueueItem(url.key, file.name);
+        $(e.target).val(null);
+      }.bind(this));
+    }.bind(this));
+  },
+
+  clearAttachment: function(e) {
+    var $target = $(e.target);
+    var key = $target.data('key');
+    var payload = this.getStorage();
+    $target.remove();
+  },
+
+  save: function(e) {
+    e.preventDefault();
+    this.setStorage();
   }
 });
 
+function getUploadUrl() {
+  return $.getJSON('/upload_proxy').then(function(resp) {
```

Comment:
Can this be passed in from the template? That way we can account for being mounted on a non-root path.